src/etc/inc/filter.inc

```diff
@@ -2974,7 +2974,12 @@ function filter_generate_user_rule($rule) {
 			if (!empty($ifliste)) {
 				$aline['interface'] = " on { {$ifliste} } ";
 			} else {
-				$aline['interface'] = "";
+				/* The rule specifies interface(s), but not one could be found.
+				 * Ignore this rule rather than generating one w/o an
+				 * 'on (interface)' specification, since that would incorrectly
+				 * apply to all interfaces!
+				 */
+				return "# rule " . $rule['descr'] . " has no matching interfaces";
```

Comment:
You're right... all of that wording was a little awkward. I changed it to "has no *enabled* interfaces" since that's even more accurate. ("Active" could imply Ethernet cable connected and green light on.)

I also simplified the comment above it.